lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4818,11 +4818,6 @@ namespace Js
 #endif
         }
 
-        if (result == FALSE && (propertyOperationFlags & PropertyOperation_StrictMode))
```

Comment:
To try and catch any errors in future, it might be worth to add an assert here along the lines of `Assert(result || !(propertyOperationFlags & PropertyOperation_StrictMode)`; that way there is no overhead in release builds, but we should catch regressions in testing.

Reply:
maybe `result || !(propertyOperationFlags & (PropertyOperation_StrictMode | PropertyOperation_ThrowOnDeleteOrNotConfig))`, since the point of this change is updating the contract elsewhere to respect both flags

---
In reply to: [173516816](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173516816) [](ancestors = 173516816)